macros/models/_sector/tokens/transfers_enrich.sql

```diff
@@ -7,33 +7,78 @@ CASE token_standard
 END
 {%- endmacro-%}
 
-{%- macro transfers_enrich(blockchain, transfers_base, native_symbol) %}
-SELECT t.blockchain
-, t.block_time
-, t.block_number
-, t.tx_hash
-, t.tx_index
-, t.evt_index
-, t.trace_address
-, t.token_standard
-, t.tx_from
-, t.tx_to
-, t."from"
-, t.to
-, t.contract_address
-, {{ case_when_token_standard("'" ~ native_symbol ~ "'", 'tokens_erc20.symbol', 'NULL') }} AS symbol
-, t.amount_raw
-, {{case_when_token_standard('t.amount_raw / power(10, 18)', 'amount_raw / power(10, tokens_erc20.decimals)', 'cast(t.amount_raw as double)')}} AS amount
-, prices.price AS usd_price
-, {{case_when_token_standard('(t.amount_raw / power(10, 18)) * prices.price',
-    '(t.amount_raw / power(10, tokens_erc20.decimals)) * prices.price',
-    'NULL')}} AS usd_amount
-FROM {{transfers_base}} t
-LEFT JOIN {{source('tokens', 'erc20')}} tokens_erc20 on tokens_erc20.blockchain = '{{blockchain}}' AND tokens_erc20.contract_address = t.contract_address
-LEFT JOIN {{ source('prices', 'usd') }} prices ON prices.blockchain = '{{blockchain}}'
+{% macro transfers_enrich(
+    base_transfers = null
+    , tokens_erc20_model = null
+    , prices_model = null
+    , evms_info_model = null
+    , transfers_start_date = '2000-01-01'
+    )
+%}
+
+WITH base_transfers as (
+    SELECT
+        *
+    FROM
+        {{ base_transfers }}
+    {% if is_incremental() %}
+    WHERE
+        {{ incremental_predicate('block_time') }}
+    {% endif %}
+)
+, prices AS (
+    SELECT
+        blockchain
+        , contract_address
+        , minute
+        , price
+    FROM
+        {{ prices_model }}
+    {% if is_incremental() %}
+    WHERE
+        {{ incremental_predicate('minute') }}
+    {% else %}
+    WHERE
+        minute >= TIMESTAMP '{{ transfers_start_date }}'
+    {% endif %}
+)
+SELECT
+    t.unique_key
+    , t.blockchain
+    , t.block_date
+    , t.block_time
+    , t.block_number
+    , t.tx_hash
+    , t.evt_index
+    , t.trace_address
+    , t.token_standard
+    , t.tx_from
+    , t.tx_to
+    , t.tx_index
+    , t."from"
+    , t.to
+    , t.contract_address
+    , {{case_when_token_standard('evms_info.native_token_symbol', 'tokens_erc20.symbol', 'NULL')}} AS symbol
+    , t.amount_raw
+    , {{case_when_token_standard('t.amount_raw / power(10, 18)', 't.amount_raw / power(10, tokens_erc20.decimals)', 'cast(t.amount_raw as double)')}} AS amount
+    , prices.price AS price_usd
+    , {{case_when_token_standard('(t.amount_raw / power(10, 18)) * prices.price',
+        '(t.amount_raw / power(10, tokens_erc20.decimals)) * prices.price',
+        'NULL')}} AS amount_usd
+FROM
+    base_transfers as t
+INNER JOIN
+    {{ evms_info_model }} as evms_info
+    ON evms_info.blockchain = t.blockchain
+LEFT JOIN
+    {{ tokens_erc20_model }} as tokens_erc20
+    ON tokens_erc20.blockchain = t.blockchain
+    AND tokens_erc20.contract_address = t.contract_address
+LEFT JOIN prices
+    ON prices.blockchain = t.blockchain
```

Comment:
it might help if this is static instead of using the transfer blockchain

Reply:
i can't even get the `base_` level to rebuild to test out running the `enrich` step. will keep this open to try once we get there though